src/topology/running.rs

```diff
@@ -444,6 +444,7 @@ impl RunningTopology {
             .filter(|&(existing_sink, _)| existing_sink)
             .map(|(_, key)| key.clone());
 
+        // TODO: also remove this for enrichment tables that act as sinks
```

Comment:
I think we would be better off disallowing buffers in front of enrichment sinks, at least to start with. They shouldn't be needed and we can avoid this particular bit of complexity.

Reply:
Right now, that would require different branches for sinks and enrichment tables, since sinks always have buffers (if I understand the code correctly). Currently I have hardcoded it to always use the default buffer (which is a single memory buffer).

That might not really be that hard to do and maybe it would be a smarter thing to do, since some of the things that sink has just don't make sense for these sinks built out of the table, but I think that brings us back to the above point, of changing `SinkOuter` usage into some smaller components.